packages/web-sdk/src/transports/otlp/transport.ts

```diff
@@ -0,0 +1,174 @@
+      if (this.disabledUntil > new Date(this.getNow())) {
+        this.logWarn(`Dropping transport item due to too many requests. Backoff until ${this.disabledUntil}`);
+
+        return Promise.resolve();
+      }
+
+      await this.promiseBuffer.add(() => {
+        const body = JSON.stringify(this.getListTransportBody(items));
+
+        const { url, requestOptions, apiKey } = this.options;
+        const { headers, ...restOfRequestOptions } = requestOptions ?? {};
+
+        return fetch(url, {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+            ...(headers ?? {}),
+            ...(apiKey ? { 'x-api-key': apiKey } : {}),
+          },
+          body,
+          keepalive: true,
+          ...(restOfRequestOptions ?? {}),
+        })
+          .then((response) => {
+            if (response.status === 429 /* too many requests */) {
+              this.disabledUntil = this.getRetryAfterDate(response);
+              this.logWarn(`Too many requests, backing off until ${this.disabledUntil}`);
+            }
+
+            return response;
+          })
+          .catch((err) => {
+            this.logError('Failed sending payload to the receiver\n', JSON.parse(body), err);
+          });
+      });
+    } catch (err) {
+      this.logError(err);
+    }
+  }
+
+  private getListTransportBody(items: TransportItem[]): OtlpTransportBody {
+    const body: OtlpTransportBody = {
+      resourceMetrics: [],
+      resourceLogs: [],
+      resourceSpans: [],
+    };
+
+    for (const item of items) {
+      const { type } = item;
+
+      if (type === TransportItemType.MEASUREMENT) {
+        body.resourceMetrics.push((item as TransportItem<MeasurementEvent>).payload);
+      }
+
+      if (type === TransportItemType.TRACE) {
+        body.resourceSpans.push((item as TransportItem<TraceEvent>).payload);
+      }
+
+      if (type === TransportItemType.LOG) {
+        body.resourceLogs.push((item as TransportItem<LogEvent>).payload);
+      }
+
+      if (type === TransportItemType.EXCEPTION) {
+        body.resourceLogs.push((item as TransportItem<ExceptionEvent>).payload);
+      }
+
+      if (type === TransportItemType.EVENT) {
+        body.resourceLogs.push((item as TransportItem<EventEvent>).payload);
+      }
+    }
+
+    return body;
+  }
+
+  private getRetryAfterDate(response: Response): Date {
+    const now = this.getNow();
+    const retryAfterHeader = response.headers.get('Retry-After');
+
+    if (retryAfterHeader) {
+      const delay = Number(retryAfterHeader);
+
+      if (!isNaN(delay)) {
+        return new Date(delay * 1000 + now);
+      }
+
+      const date = Date.parse(retryAfterHeader);
+
+      if (!isNaN(date)) {
+        return new Date(date);
+      }
+    }
+
+    return new Date(now + this.rateLimitBackoffMs);
+  }
+}
+
+interface OtlpTransportBody {
+  resourceMetrics: MeasurementEvent[];
+  resourceLogs: (LogEvent | ExceptionEvent | EventEvent)[];
```

Comment:
Not sure that there are typescript types for otlp logs yet :/ might need to be generated from protobuf